examples/simple-grep/README.md

```diff
@@ -0,0 +1,129 @@
+There is a basic example usage of Iris and a tutorial-like explanation:
+
+First of all, let's create a newtype wrapper for the main monad of Iris:
+
+```haskell
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+
+module Main (main) where
+
+import Prelude hiding (readFile)
+import Control.Monad.IO.Class (MonadIO (..))
+import Control.Monad.Reader (MonadReader)
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.ByteString as BS
+import qualified Data.Text.Lazy as T
+import qualified Data.Text.Lazy.Encoding as TLE
+import qualified Options.Applicative as Opt
+
+import qualified Colourista
+import qualified Iris
+import qualified Paths_iris as Autogen
+
+newtype App a = App
+    { unApp :: Iris.CliApp SimpleGrep () a
+    } deriving newtype
+        ( Functor
+        , Applicative
+        , Monad
+        , MonadIO
+        , MonadReader (Iris.CliEnv SimpleGrep ())
+        )
+```
+
+It uses `Iris.CliApp SimpleGrep () a` that means `CliApp` will consume `SimpleGrep` data of commands for CLI description. Also, there's `()` in the second argument, it could be a specific environment type for our application, but in our case, it isn't necessary.
+`SimpleGrep` is described below:
+
+```haskell
+data SimpleGrep = SimpleGrep
+    {
+      filePath :: !String
+    , substring :: !String
+    }
+```
+
+After the creation of a base monad with data-type of commands, let's instantiate settings to Iris:
+
+```haskell
+appSettings :: Iris.CliEnvSettings SimpleGrep ()
+appSettings = Iris.defaultCliEnvSettings
+    { Iris.cliEnvSettingsHeaderDesc = "Iris usage example"
+    , Iris.cliEnvSettingsProgDesc = "A simple grep utility"
+    , Iris.cliEnvSettingsVersionSettings =
+        Just (Iris.defaultVersionSettings Autogen.version)
+            { Iris.versionSettingsMkDesc = \v -> "Simple grep utility v" <> v
+            }
+    , Iris.cliEnvSettingsRequiredTools = []
+
+    , Iris.cliEnvSettingsCmdParser = SimpleGrep
+        <$> Opt.strOption
+            (  Opt.long "filePath"
+            <> Opt.short 'f'
+            <> Opt.metavar "PATH"
+            <> Opt.help "filePath to find and grep file"
+            )
+        <*> Opt.strOption
+            (  Opt.long "substring"
+            <> Opt.short 's'
+            <> Opt.help "Substring to find and highlight"
+            )
```

Comment:
Let's move CLI parser into a separate section combined with the `SimpleGrep` definition so the `appSettings` definition will be even shorter